blocks/library/embed/index.js

```diff
@@ -26,7 +27,33 @@ import BlockAlignmentToolbar from '../../block-alignment-toolbar';
 // These embeds do not work in sandboxes
 const HOSTS_NO_PREVIEWS = [ 'facebook.com' ];
 
-function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, keywords = [] } ) {
+// Caches the embed API calls, so if blocks get transformed, or deleted and added again, we don't spam the API.
+const wpEmbedAPI = memoize( ( url ) => wp.apiRequest( { path: `/oembed/1.0/proxy?${ stringify( { url } ) }` } ) );
+
+const matchesPatterns = ( url, patterns ) => {
+	for ( const patternIndex in patterns ) {
+		const pattern = new RegExp( '^https?:\/\/' + patterns[ patternIndex ] + '$', 'i' );
```

Comment:
Rather than constructing these regular expressions, should this just be duplicated into all of the patterns below? Not sure it's obvious to an implementer that they need to: provide string which is a regular expression, but not including protocol, or end anchor, or flags.

What about providers which explicitly don't support HTTPS? 

https://github.com/WordPress/wordpress-develop/blob/3a46cba/src/wp-includes/class-oembed.php#L63-L64

I can appreciate the effort to DRY this, but to me it causes more confusion to the maintainer than it helps.